papers/ring_vrf/pedersen_vrf.tex

```diff
@@ -23,30 +25,32 @@ \section{Ring VRF construction}% {Ring VRFs from the Pedersen VRF}
  but highlight several points: 
 % We shall need $\ecE$ to be pairing friendly in later sections, but ignore this here.
 %
-Although non-essential, we prefer if \PreOut cannot be verified
-by pairings, as a miss-use resistance measure.
-We therefore suggest a ``sister'' Edwards curve $\ecE'$ with
- a subgroup $\grE'$ of the same order $p$ as $\grE_1$,
- and cofactor $h'$ divisible by 4.
-You may take $\ecE' = \ecE_1$ and $h' = 1$ if reading only for
-security proofs, but this incurs risks in deployments.
-%
-As Groth16 dominates our ring VRF verification costs, we describe only
-the non-batchable \PedVRF analogous to \cite{nsec5,VXEd25519,draft-irtf-cfrg-vrf-10},
-and ignore the cofactor $h_1$ of $\grE_1$ of $\ecE_1$, 
-but caution \cite{HdVBatchEd25519} impacts batch verifiable flavors.
-%
-We select a generator $\genG$ of $\grE$ as our public key base point
- by any desired method, but then
-fix a second generator $\genB$ of $\grE$ independent from $\genG$.
+%COMMENT: I put these in the end. 
+%Although non-essential, we prefer if \PreOut cannot be verified
+%by pairings, as a miss-use resistance measure.
+%We therefore suggest a ``sister'' Edwards curve $\ecE'$ with
+% a subgroup $\grE'$ of the same order $p$ as $\grE_1$,
+% and cofactor $h'$ divisible by 4.
+%You may take $\ecE' = \ecE_1$ and $h' = 1$ if reading only for
+%security proofs, but this incurs risks in deployments.
+%%
+%As Groth16 dominates our ring VRF verification costs, we describe only
+%the non-batchable \PedVRF analogous to \cite{nsec5,VXEd25519,draft-irtf-cfrg-vrf-10},
+%and ignore the cofactor $h_1$ of $\grE_1$ of $\ecE_1$, 
+%but caution \cite{HdVBatchEd25519} impacts batch verifiable flavors.
+%%
+%We select a generator $\genG$ of $\grE$ as our public key base point
+% by any desired method, but then
+%fix a second generator $\genB$ of $\grE$ independent from $\genG$.
 
 We now describe the Pedersen VRF \PedVRF evaluation proof used by our ring VRF:
 %
 \KeyGen works exactly like EC VRF, but
  \Eval differs by not injecting \pk into \msg:
```

Comment:
It was never quite clear to me what is meant by injecting \pk into \msg. Do previous VRFs really do that?

Reply:
I think injecting pk means that the ring VRF (which is H in our case) does not has pk as an input. In ECVRF, pk is the part of the input of H. Here, we do not include pk because the verifier should not need to know pk to obtain the evaluation value in ring VRF.